docs/design/README.md

```diff
@@ -0,0 +1,552 @@
+
+A new directory will be added: `kits`.
+This will be a Cargo workspace with one Cargo package for each kit.
+
+```
+kits
+├── aws
+│   ├── build.rs
+│   ├── Cargo.toml
+│   └── lib.rs
+├── core
+│   ├── build.rs
+│   ├── Cargo.toml
+│   └── lib.rs
+├── ecs
+│   ├── build.rs
+│   ├── Cargo.toml
+│   └── lib.rs
+├── k8s
+│   ├── build.rs
+│   ├── Cargo.toml
+│   └── lib.rs
+├── metal
+│   ├── build.rs
+│   ├── Cargo.toml
+│   └── lib.rs
+└── vmware
+    ├── build.rs
+    ├── Cargo.toml
+    └── lib.rs
+```
+
+A variant such as `metal-k8s-1.22` would consume the necessary kits.
+For example, it would reference `core-kit`, `k8s-kit` and `metal-kit`.
+To add `iam-authenticator`, `aws-kit` would be added.
+
+It may be convenient for certain build steps to have a single package representing a kit, such as `bottlerocket-aarch64-core-kit-0.1-1.aarch64.rpm`
+If so, these spec files could be added to the above directory tree.
+It is also possible to generate these spec files as they would be simple aggregations of all the packages in the kit.
+
+### Kit Structure
+
+A kit container will be built using `FROM scratch` because it does not need a runtime environment.
+It's directory structure will look like this:
+
+```sh
+# tree /local --dirsfirst
+/local
+├── etc
+│   └── yum.repos.d
+│       └── bottlerocket-core-kit.repo
+└── kits
+    └── bottlerocket-core-kit
+        ├── archives
+        │   └── bottlerocket-x86_64-kernel-5.15-archive-5.15.102-1.x86_64.rpm
+        ├── bootconfig
+        │   ├── empty-bootconfig.data
+        │   └── qemu-x86-console-bootconfig.data
+        ├── licenses
+        │   ├── COPYRIGHT
+        │   ├── LICENSE-APACHE
+        │   └── LICENSE-MIT
+        ├── migrations
+        │   └── bottlerocket-x86_64-migrations-0.0-0.x86_64.rpm
+        ├── repodata
+        │   ├── 30e098715e13cbe4733c6b4518dcc2cdc68d1ac3277eefad2ee5ac171c4b9023-primary.sqlite.bz2
+        │   ├── 5a22c22832fa81e55f84de715cf910cebc64d472be65a7124fd8b02ff7c777fb-filelists.xml.gz
+        │   ├── 719d1eef5ce4d194d4c5c51171dce8987cb21e88338f1d5dff7554dacabf1004-other.sqlite.bz2
+        │   ├── 7cbab1a2f05c1eda86d3d125497af680d9100665e3eb30c7560926eabbc2c655-primary.xml.gz
+        │   ├── d61fe62ca166eab1cbd502cf21cd977798ef348de49fb35e38a93ebf696eb0a5-filelists.sqlite.bz2
+        │   ├── fcd4dadc59c6d6512410b103a38762bf266389e01a60add12319191c4cc16ec4-other.xml.gz
+        │   └── repomd.xml
+        ├── bottlerocket-x86_64-acpid-2.0.34-1.x86_64.rpm
+        ├── bottlerocket-x86_64-apiclient-0.0-0.x86_64.rpm
+        ├── ...
+        └── bottlerocket-x86_64-wicked-0.6.68-1.x86_64.rpm
+```
+
+The file at `/local/etc/yum.repos.d` can be placed at `/etc/yum.repos.d` later when multiple kits are being aggregated.
+This will alow `dnf` to treat each kit as a yum repository.
+When aggregating kits, the yum repos will be prioritized according to the maintainer's requirements.
+
+The presence of `bootconfig` is questionable.
```

Comment:
Will different out of tree builds have different boot configs? Maybe it make sense to have the bootconfig as it's own kit if we anticipate out of tree builders to modify/make their own bootconfigs.